homeassistant/config_entries.py

```diff
@@ -750,30 +750,51 @@ def async_update_entry(
         data: dict = _UNDEF,
         options: dict = _UNDEF,
         system_options: dict = _UNDEF,
-    ) -> None:
-        """Update a config entry."""
-        if unique_id is not _UNDEF:
+    ) -> bool:
+        """Update a config entry.
+
+        If the entry was changed, the update_listeners are
+        fired and this function returns True
+
+        If the entry was not changed, the update_listeners are
+        not fired and this function returns False
+        """
+        changed = False
+
+        if unique_id is not _UNDEF and entry.unique_id != unique_id:
+            changed = True
             entry.unique_id = cast(Optional[str], unique_id)
 
-        if title is not _UNDEF:
+        if title is not _UNDEF and entry.title != title:
+            changed = True
             entry.title = cast(str, title)
 
-        if data is not _UNDEF:
+        if data is not _UNDEF and dict(entry.data) != dict(data):
+            changed = True
             entry.data = MappingProxyType(data)
 
-        if options is not _UNDEF:
+        if options is not _UNDEF and dict(entry.options) != dict(options):
+            changed = True
             entry.options = MappingProxyType(options)
 
-        if system_options is not _UNDEF:
+        if system_options is not _UNDEF and entry.system_options.as_dict() != dict(
```

Comment:
This is not a right check, we need to check if the dict is a subset of the other one. 

We can do that with:

```python
system_options.items() <= entry.system_options.items()
```

Reply:
hmm `AttributeError: 'SystemOptions' object has no attribute 'items'`

I think a simple compare might work here. Will try that without the copy and increase the coverage